sql/branch.sql

```diff
@@ -1,4 +1,35 @@
-ALTER TABLE payment_instructions ADD COLUMN participant_id bigint DEFAULT NULL
-   REFERENCES participants(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
-ALTER TABLE payment_instructions ADD COLUMN team_id bigint DEFAULT NULL
-   REFERENCES teams(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
+-- https://github.com/gratipay/gratipay.com/pull/4046
+
+BEGIN;
+
+    ALTER TABLE payment_instructions ADD COLUMN participant_id bigint DEFAULT NULL
+       REFERENCES participants(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
+    ALTER TABLE payment_instructions ADD COLUMN team_id bigint DEFAULT NULL
+       REFERENCES teams(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
+
+    DROP VIEW current_payment_instructions;
+
+    UPDATE payment_instructions AS pi
+       SET participant_id = p.id
+      FROM participants AS p
+     WHERE p.username = pi.participant
+       AND pi.participant_id IS NULL;
+
+    UPDATE payment_instructions AS pi
+       SET team_id = t.id
+      FROM teams AS t
+     WHERE t.slug = pi.team
+       AND pi.team_id IS NULL;
+
+    ALTER TABLE payment_instructions DROP COLUMN participant;
+    ALTER TABLE payment_instructions DROP COLUMN team;
+
+    ALTER TABLE payment_instructions ALTER COLUMN participant_id SET NOT NULL;
+    ALTER TABLE payment_instructions ALTER COLUMN team_id SET NOT NULL;
+
+    CREATE VIEW current_payment_instructions AS
+        SELECT DISTINCT ON (participant_id, team_id) *
+          FROM payment_instructions
+      ORDER BY participant_id, team_id, mtime DESC;
+
+END;
```

Comment:
> Can we club the two `UPDATE`s into a single `UPDATE` statement so that this takes lesser time?

Done in 7a0c2d9902d81da38de4b2eb3a5a18aa882e7947, and it looks like indeed it's twice as fast (even though, for simplicity, it uses a subselect instead of a `FROM`):

```
[gratipay] $ time psql gratipay-bak < sql/branch.sql
Null display is "¤".
Line style is unicode.
Border style is 2.
BEGIN
ALTER TABLE
ALTER TABLE
DROP VIEW
UPDATE 3787
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
CREATE VIEW
CREATE TRIGGER
COMMIT

real    0m0.827s
user    0m0.005s
sys     0m0.005s
[gratipay] $
```